core/src/main/scala/org/typelevel/paiges/Doc.scala

```diff
@@ -567,11 +567,12 @@ object Doc {
    */
   private[paiges] case class Align(doc: Doc) extends Doc
 
-  /**
-   * Invariant: generate is referentially transparent
-   */
-  private[paiges] case class LazyDoc(private val generate: () => Doc) extends Doc {
-    lazy val evaluated: Doc = generate()
+  private[paiges] case class LazyDoc(thunk: LazyDoc.Thunk) extends Doc
+
+  private[paiges] object LazyDoc {
+    class Thunk(value: => Doc) {
```

Comment:
can we not nest this and just put it inside `object Doc`?

Reply:
Done.